### PR TITLE
Implements LaneGeometry::WDot and LaneGeometry::WInverse.

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

Part of #7 

## Summary
Provides `WDot` and `WInverse` methods.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

### CODE REVIEW

File: src/geometry/lane_geometry.cc
Comment (agalbachicar) [inline src/geometry/lane_geometry.cc:114]: Because we know that the orientation remains the same at any r and h for a given p. You can get the columns of the rotation matrix and get each direction vector. In other words, I suggest you pick the Orientation(closest_centerline_point.p) and get the second column to build r_hat and the third column to build h_hat.

File: src/geometry/utility/geometry.cc
Comment (agalbachicar) [inline src/geometry/utility/geometry.cc:353]: Consider using Vector::nomalized(). Also, you could use it in place and save you from declaring the variable for that purpose only.

File: src/geometry/utility/geometry.cc
Comment (agalbachicar) [inline src/geometry/utility/geometry.cc:358]:
```suggestion
  const maliput::math::Vector3 point = InterpolatedPointAtP(LineString3d{segment.first, segment.second}, p);
```

**Review by agalbachicar — CHANGES_REQUESTED** [review]
Just a few minor comments. PTAL.

File: src/geometry/utility/geometry.cc
Comment (francocipollone) [reply src/geometry/utility/geometry.cc]: Done.

File: src/geometry/utility/geometry.cc
Comment (francocipollone) [reply src/geometry/utility/geometry.cc]: Right!
Done!

File: src/geometry/lane_geometry.cc
Comment (francocipollone) [reply src/geometry/lane_geometry.cc]: You are right, I can directly use the orientation matrix to get r and h.
Done.

File: include/maliput_sparse/geometry/utility/geometry.h
Comment (francocipollone) [reply include/maliput_sparse/geometry/utility/geometry.h]: Done.

File: include/maliput_sparse/geometry/utility/geometry.h
Comment (francocipollone) [reply include/maliput_sparse/geometry/utility/geometry.h]: Done.

**Review by agalbachicar — APPROVED** [review]
LGTM

**Comment by francocipollone** [comment]
I just noticed that I mistakenly merged [this ](https://github.com/maliput/maliput_sparse/pull/16) into this branch.

**Comment by francocipollone** [comment]
I will merge it to main once this is approved again @agalbachicar  :+1:

**Pull request merged**
